### ISSUE TITLE
Is there any way to trim whitespaces from path, query and body parameters before binding?

### ISSUE THREAD

**Issue by therox** [gin-gonic/gin#3350]
If user accidentally adds whitespaces in param
ex.:
a := struct {
    DateField time.Time `json:"date_field" binding:required`
}

```
{
    "date_field": "2022-06-02   "
}
```
is there any way to remove extra spaces before binding to struct to prevent possible errors?

**Comment by 454532358** [comment]
you have to write it like this
`a := struct {
    DateField time.Time `json:"date_field" binding:"required" time_format:"2006-01-02" time_utc:"1"`
}`

**Comment by dansimau** [comment]
I also have the same requirement.

This can be achieved with custom bindings, but I've found myself copy/pasting a lot of internal code just to recreate the builtin binders but with a TrimSpace.

Maybe it would make more sense to be able to hook into the existing binding via pre/post hooks.

**Comment by therox** [comment]
Or it could be something like `strip_whitespace=True` in pydantic.

**Comment by Cookiery** [comment]
Maybe you can handle it in the frontend.

**Comment by jledesma84** [comment]
@therox did you find anything interesting to solve this?

**Comment by therox** [comment]
@jledesma84 No, I didn't. I have to use reflect manually to get access to structure field's data.
